pkgs/development/compilers/cudatoolkit/redist/build-cuda-redist-package.nix

```
@@ -4,21 +4,58 @@
 , fetchurl
 , autoPatchelfHook
 , autoAddOpenGLRunpathHook
+, manifestAttribute ? null
 }:
 
 pname:
 attrs:
 
 let
-  arch = "linux-x86_64";
+  systemToManifestAttribute = {
+    "x86_64-linux" = [ "linux-x86_64" ];
+    "aarch64-linux" = [ "linux-aarch64" "linux-sbsa" ];
+    "powerpc64le-linux" = [ "linux-ppc64le" ];
+  };
+  systemToManifestAttributeOrDefault = default: system:
+    if builtins.hasAttr system systemToManifestAttribute && builtins.length systemToManifestAttribute.${system} > 0
+    then builtins.head systemToManifestAttribute.${system}
+    else default; # Avoids evaluation errors on missing attributes
+  isSystemSupported = system: builtins.any (at: builtins.hasAttr at attrs) systemToManifestAttribute.${system};
+  supportedPlatforms = builtins.concatMap
+    (system: if isSystemSupported system then [ system ] else [ ])
+    (builtins.attrNames systemToManifestAttribute);
+
+  inherit (stdenv) system;
+
+  renames = {
+    manifestAttribute =
+      if manifestAttribute == null
+      then systemToManifestAttributeOrDefault "linux-x86_64" system
+      else manifestAttribute;
+  };
 in
+let
+  inherit (renames) manifestAttribute;
```

```suggestion
  manifestAttribute =
    if manifestAttribute == null
    then systemToManifestAttributeOrDefault "linux-x86_64" stdenv.system
    else manifestAttribute;
```

why 2 separate `let ... in` clauses?

---

To avoid potential infinite recursion depending on Nix version

---

Replaced with `{ ... }@inputs: ...`